### ISSUE TITLE
please add a config option that allows setting the ACME server CA certificate(s)

### ISSUE THREAD

**Issue by calestyo** [certbot/certbot#9610]
Hey.

Please add a command (and config file option) that allows to set the ACME server CA certificate(s) **per ACME account**.<br> Ideally this should allow to set more than just one CA certificate (if the ACME server's CA uses e.g. RSA or ECDSA certs).

The main reason for this is IMO, if one doesn't want to add the server's CA to the system's default CA certificate store (i.e. `/etc/ssl/certs`).<br>
The reason for that in turn is security, because any other tools would per default trust the CAs in that store, which may not be desired.

Others seem to have wanted it for use with a ACME server that uses a self-signed cert (though I personally don't think that this is an original reason, as one could simply add that to the CA store).

This feature has been asked for previously (in #9100 which seems to have been closed due to lack of progress) and so far it seems the "solution" was only to improve documentation about `REQUESTS_CA_BUNDLE`.<br>
`REQUESTS_CA_BUNDLE` is however not really a solution because in the `certbot renew`-case it would apply to **all** while one would typically want it to apply only to a specific ACME account (consider multiple servers being used).

Thanks,
Chris.

**Comment by bmw** [comment]
I understand your request and use case, but I'm going to close this as wontfix as I think it's a feature very few people would actually use. If you need this, you can do something like use custom configuration directories for each ACME server by setting `--config-dir` on the command line and configure `REQUESTS_CA_BUNDLE=... certbot renew --config-dir ...` to run for each of these and/or make use of `--cert-name` to limit which certificates `certbot renew` will attempt to renew.

**Comment by calestyo** [comment]
Hey.

Well in the end it's of course up to what you guys want... but I figured such an option would be really pretty small (one option in parsing, and setting the respective parameter for Python `requests`).

IMO, the whole TLS ecosystem is anyway inherently broken. Not by the algos themselves but by the common practise of simply blindly trusting a CA bundle for basically everything.<br>
That bundle typically contains something around 150 root CAs, quite a few of them from totalitarian countries and from CAs which in the past have already been caught at forging certs (of course these were aaaalways just accidents).<br>
In addition to that, many of the root CAs have a completely opaque number of intermediate CAs which  also can issue more or less what they want.

I'm not saying that I have a solution for the root problem, but things would at least be better, if systems hardcode the CA they expect - at least in cases where communication happens only with the same remotes (and if those use certs by the same CA)…<br>
… which I think should typically be the case for any ACME server.

Since letsencrypt/certbot aim to improve security, I figured it wouldn't harm if they set a good example and make it easy to specify the trusted CA(s).


Your workaround with `--config-dir` is actually better than what I do now (setting `REQUESTS_CA_BUNDLE=` in a systemd override for the whole service)... but it still has the disadvantage that it will require one to override the distro shipped cronjob or systemd service, which in turn means: not "easy" to use.

Anyway ...

Cheers,
Chris.
